analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/UsingNonstandardCryptographyBase.cs

```diff
@@ -30,10 +30,10 @@ public abstract class UsingNonstandardCryptographyBase<TSyntaxKind, TTypeDeclara
         where TTypeDeclarationSyntax : SyntaxNode
         where TSyntaxKind : struct
     {
-        internal const string DiagnosticId = "S2257";
-
-        protected const string MessageFormat = "Make sure using a non-standard cryptographic algorithm is safe here.";
+        private const string DiagnosticId = "S2257";
+        private const string MessageFormat = "Make sure using a non-standard cryptographic algorithm is safe here.";
 
+        private readonly DiagnosticDescriptor rule;
         private readonly ImmutableArray<KnownType> nonInheritableClassesAndInterfaces =
```

Comment:
Could be made static. Are there any rules, on which resources should be made static?

Reply:
If it can be `static` then it should be. It makes sense in most of the cases.

Reply:
FunFact: That rule is in the demo too.